### ISSUE TITLE
Compound heterozygous gene in father and child not prioritized in trio analysis

### ISSUE THREAD

**Issue by oleraj** [exomiser/Exomiser#368]
Hi,

I have a trio where the father and daughter are both compound heterozygous for the known causal gene (autosomal) but Exomiser is not prioritizing the gene.  When I run with the daughter alone, the causal gene is ranked number 4 based on the phenotype with a combined Exomiser score of 0.9032 but when I add the parents with father as affected and mom as unaffected it goes down to rank 75 with a score of 0.2864.  If I change the affected status of both parents to *unknown*, the gene is ranked 154, still with a score of 0.2864.  When I run it as a singleton, it's prioritized under the AR model but when I add parents it comes through the AD model, which is probably why it gets a hit in the score.  I was most surprised that setting the father as *unknown* still didn't allow the gene to be prioritized with the AR model.  Probably not something you see frequently where an AR disorder appears to be inherited as AD, however this is a real case so I'm wondering if it's possible to support this type of scenario in a trio format, or how you would recommend handling this.  

The genetics:
|  Position | Ref | Alt | Father (aff) | Mother (unaff) | Daughter (aff)| 
| -----| ----| -------| -------| ----------| --- | 
| Position 1 | G | A | G/***A*** | G/***A*** | G/***A*** | 
| Position 2 | G | A | G/***A*** | G/G | G/***A*** | 

Thanks,

Andrew

**Comment by damiansm** [comment]
Hi Andrew,

We use Jannovar as the library for variant annotation and segregation filtering so we are going to have to delve into what is happening in there. @pnrobinson and @holtgrewe any ideas?

Best wishes
Damian

**Comment by pnrobinson** [comment]
@oleraj  So basically this is a case of pseudodominant inheritance? Still, given the genotype this should be highly ranked. @holtgrewe should we add a unit test to ensure that this works? Do you know the current status here?

**Comment by oleraj** [comment]
@pnrobinson Yes, it appears so.  I wasn't familiar with the term until you mentioned it but that describes this.  For additional information, it looks like there is some level of consanguinity in this family (~6% IBD so somewhere between first and second cousins), which is why both the mother and father have the same rare pathogenic allele.  And at least one of their children is homozygous for the shared pathogenic allele as well, which would also follow the pseudodominant pattern though the genotype combination in the father (compound het of 2 alleles) would be different from the child (homozygous alt of one allele).

**Comment by oleraj** [comment]
I'll just note that we identified another case where the likely candidate gene exhibits pseudodominant inheritance so it would be great if this could be detected in Exomiser.  This second case follows the same pattern - i.e., one parent is homozygous, the other parent is heterozygous and the affected children are compound heterozygous.

**Comment by pnrobinson** [comment]
@julesjacobsen I think that the Jannovar autosomal recessive filter should already work for this -- but we should make sure and add some tests if this is not yet available. @oleraj  Did you try to identify this case using the autosomal recessive filter?

**Comment by oleraj** [comment]
@pnrobinson 
Oh, actually, I misspoke -- this latter example is _not_ the same pattern.  For the latter case, I don't actually have access to the exome data for that one yet but this was through a collaborator that the candidate gene exhibited this pattern.  I just wanted to mention it as another example to support this issue.  One could make a test case for this scenario as well.

For the first case I mentioned (father comp_het, mother het, affected child comp_het) if I use the proband alone, the causal gene is identified in the autosomal recessive filter, ranked #4 based on the phenotypes supplied (using v.12.1.0).  If I include the parents it doesn't show up at all in the autosomal recessive filter.  With the parents it does show up in the "ANY" inheritance model -- ranked #10 -- but ideally we should be able to see this in the autosomal recessive model with parents.

**Comment by julesjacobsen** [comment]
@oleraj  what was the frequency of the variants? 

@pnrobinson I think there might be an issue with the Jannovar inheritance mode analysis. Will need to look into this further.

**Comment by julesjacobsen** [comment]
@oleraj., @pnrobinson  I set-up a test using these alleles and the Inheritance mode analysis set to include AD, AR and ANY. Looks like Jannovar is not working as intended?

Multi-sample father hom alt, allele 1:
```
Built position 1 [VC null @ 1:11111 Q0.00 of type=SNP alleles=[G*, A] attr={} GT=[[Daughter G*/A],[Father A/A],[Mother G*/A]] filters=
Built position 2 [VC null @ 1:22222 Q0.00 of type=SNP alleles=[G*, A] attr={} GT=[[Daughter G*/A],[Father G*/A],[Mother G*/G*]] filters=
2 passed alleles in gene
VariantEvaluation{assembly=hg19 chr=1 strand=+ start=11111 end=11111 length=1 ref=G alt=A qual=0.0 SNV SEQUENCE_VARIANT score=0.0 PASSED failedFilters=[] passedFilters=[FREQUENCY_FILTER] compatibleWith=[] sampleGenotypes={Daughter=0/1, Mother=0/1, Father=1/1}}
VariantEvaluation{assembly=hg19 chr=1 strand=+ start=22222 end=22222 length=1 ref=G alt=A qual=0.0 SNV SEQUENCE_VARIANT score=0.0 PASSED failedFilters=[] passedFilters=[FREQUENCY_FILTER] compatibleWith=[] sampleGenotypes={Daughter=0/1, Mother=0/0, Father=0/1}}
Gene compatible with: [ANY]
1	11111	G	A	0	0/1:0/1:1/1	compatibleWith=[ANY]
1	22222	G	A	0	0/1:0/0:0/1	compatibleWith=[ANY]
```

Multi-sample, father comp het:
```
Built position 1 [VC null @ 1:11111 Q0.00 of type=SNP alleles=[G*, A] attr={} GT=[[Daughter G*/A],[Father G*/A],[Mother G*/A]] filters=
Built position 2 [VC null @ 1:22222 Q0.00 of type=SNP alleles=[G*, A] attr={} GT=[[Daughter G*/A],[Father G*/A],[Mother G*/G*]] filters=
2 passed alleles in gene
VariantEvaluation{assembly=hg19 chr=1 strand=+ start=11111 end=11111 length=1 ref=G alt=A qual=0.0 SNV SEQUENCE_VARIANT score=0.0 PASSED failedFilters=[] passedFilters=[FREQUENCY_FILTER] compatibleWith=[] sampleGenotypes={Daughter=0/1, Mother=0/1, Father=0/1}}
VariantEvaluation{assembly=hg19 chr=1 strand=+ start=22222 end=22222 length=1 ref=G alt=A qual=0.0 SNV SEQUENCE_VARIANT score=0.0 PASSED failedFilters=[] passedFilters=[FREQUENCY_FILTER] compatibleWith=[] sampleGenotypes={Daughter=0/1, Mother=0/0, Father=0/1}}
Gene compatible with: [ANY]
1	11111	G	A	0	0/1:0/1:0/1	compatibleWith=[ANY]
1	22222	G	A	0	0/1:0/0:0/1	compatibleWith=[ANY]
```

Proband only:
```
Built position 1 [VC null @ 1:11111 Q0.00 of type=SNP alleles=[G*, A] attr={} GT=[[Daughter G*/A]] filters=
Built position 2 [VC null @ 1:22222 Q0.00 of type=SNP alleles=[G*, A] attr={} GT=[[Daughter G*/A]] filters=
2 passed alleles in gene
VariantEvaluation{assembly=hg19 chr=1 strand=+ start=11111 end=11111 length=1 ref=G alt=A qual=0.0 SNV SEQUENCE_VARIANT score=0.0 PASSED failedFilters=[] passedFilters=[FREQUENCY_FILTER] compatibleWith=[] sampleGenotypes={Daughter=0/1}}
VariantEvaluation{assembly=hg19 chr=1 strand=+ start=22222 end=22222 length=1 ref=G alt=A qual=0.0 SNV SEQUENCE_VARIANT score=0.0 PASSED failedFilters=[] passedFilters=[FREQUENCY_FILTER] compatibleWith=[] sampleGenotypes={Daughter=0/1}}
Gene compatible with: [AUTOSOMAL_DOMINANT, AUTOSOMAL_RECESSIVE, ANY]
1	11111	G	A	0	0/1	compatibleWith=[AUTOSOMAL_DOMINANT, AUTOSOMAL_RECESSIVE, ANY]
1	22222	G	A	0	0/1	compatibleWith=[AUTOSOMAL_RECESSIVE, ANY]
```
